crates/uv/src/lib.rs

```diff
@@ -42,8 +44,45 @@ pub(crate) mod printer;
 pub(crate) mod settings;
 pub(crate) mod version;
 
+/// Resolves the script target for a run command.
+/// If it's a local file, does nothing. If it's a URL, downloads the content
+/// to a temporary file and updates the command. Prioritizes local files over URLs.
+/// Returns Some(NamedTempFile) if a remote script was downloaded, None otherwise.
+async fn resolve_script_target(
+    command: &mut ExternalCommand,
+) -> Result<Option<tempfile::NamedTempFile>> {
+    let Some(target) = command.get_mut(0) else {
+        return Ok(None);
+    };
+
+    if Path::new(target).is_file() {
+        return Ok(None);
+    }
+
+    let maybe_url = target.to_string_lossy();
+    if maybe_url.starts_with("http://") || maybe_url.starts_with("https://") {
```

Comment:
I would use an early return here. So if the target _doesn't_ look like a URL, then `return Ok(None);`. This keeps rightward drift in check and makes it easy to see under what conditions the function bails.

Reply:
Good idea.